### PR TITLE
Fix 560

### PR DESCRIPTION
**Pull request by MonkeyCanCode**
The issue arises because the second curl command does not return the correct status code due to the behavior of `getOrCreateSessionSupplier`. When a realm is not found, the method triggers realm initialization. However, realm initialization only partially bootstraps the setup (it creates the sequence object with default and populated tables but no data). As a result, when the second curl command is executed, it attempts credential validation instead of the expected check, since the schema for the required tables is created during the realm initialization.

To ensure consistent error handling and avoid premature credential validation, this PR adds an additional check to verify that the realm is fully bootstrapped before proceeding with credential validation.

### CODE REVIEW

**Comment by snazy** [comment]
Thanks @MonkeyCanCode. However, after applying this fix (and tentative ones for #561 + #562), I'm not sure whether it's worth to fix it now. I think it makes more sense to wait until #469 has landed, because it will clean up a bunch of things. Not sure about this particular one, but let's wait and see what @adutra thinks.

File: polaris-core/src/main/java/org/apache/polaris/core/persistence/LocalPolarisMetaStoreManagerFactory.java
Comment (adutra) [inline polaris-core/src/main/java/org/apache/polaris/core/persistence/LocalPolarisMetaStoreManagerFactory.java:144]:
```suggestion
      initializeForRealm(realmContext);
    }
    checkPolarisServiceBootstrappedForRealm(
        realmContext, metaStoreManagerMap.get(realmContext.getRealmIdentifier()));
```

**Review by adutra — APPROVED** [review]

**Comment by adutra** [comment]
> Thanks @MonkeyCanCode. However, after applying this fix (and tentative ones for #561 + #562), I'm not sure whether it's worth to fix it now. I think it makes more sense to wait until #469 has landed, because it will clean up a bunch of things. Not sure about this particular one, but let's wait and see what @adutra thinks.

I'd say let's get this in, as I think the race condition would be possible even after the switch to Quarkus (the switch doesn't change the logic in `LocalPolarisMetaStoreManagerFactory`.

**Pull request merged**
